### PR TITLE
[NPU] Add `mixed_precision` for Qwen2 7B

### PR DESCRIPTION
**Pull request by Oscilloscope98**
## Description

https://github.com/analytics-zoo/nano/issues/1633#issuecomment-2363009566

Support `mixed_precision`  in `from_pretrained` function for NPU
- If `mixed_precision=True` and `load_in_low_bit='sym_int4'`, Qwen2 7B will use INT8 for `lm_head`
- Model saved with `mixed_precision=True`/`False` will keep the same option when `load_low_bit` the saved model
- Disable `lm_head` split when `load_in_low_bit='sym_int8'`
- Update example accordingly

### CODE REVIEW

**Review by jason-dai — APPROVED** [review]
LGTM

File: python/llm/example/NPU/HF-Transformers-AutoModels/LLM/README.md [hunk 1]
```diff
@@ -137,29 +137,35 @@ If you encounter `TypeError: can't convert meta device type tensor to numpy. Use
...
 
-# to run Baichuan2-7B-Chat
+:: to run Baichuan2-7B-Chat
 python baichuan2.py --disable-transpose-value-cache
 ```
 
+For [Qwen2-7B](./qwen2.py), you could also try to enable mixed precision optimization for better outputs:
+
+```cmd
+python --repo-id-or-model-path Qwen/Qwen2-7B-Instruct qwen2.py
```
Comment (rnwang04) [inline python/llm/example/NPU/HF-Transformers-AutoModels/LLM/README.md:166]: python qwen2.py --repo-id-or-model-path Qwen/Qwen2-7B-Instruct  --mixed-precision

**Review by rnwang04 — APPROVED** [review]
others LGTM

File: python/llm/example/NPU/HF-Transformers-AutoModels/LLM/README.md [hunk 2]
```diff
@@ -137,29 +137,35 @@ If you encounter `TypeError: can't convert meta device type tensor to numpy. Use
...
 
-# to run Baichuan2-7B-Chat
+:: to run Baichuan2-7B-Chat
 python baichuan2.py --disable-transpose-value-cache
 ```
 
+For [Qwen2-7B](./qwen2.py), you could also try to enable mixed precision optimization when encountering output problems:
+
+```cmd
+python --repo-id-or-model-path Qwen/Qwen2-7B-Instruct qwen2.py --mixed-precision
```
Comment (rnwang04) [inline python/llm/example/NPU/HF-Transformers-AutoModels/LLM/README.md:166]: I think qwen2.py should be put before --repo-id-or-model-path  ?

**Pull request merged**